src/models/apis.rs

```diff
@@ -1,51 +1,159 @@
 use crate::models::Ratelimit;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 
+/// An outgoing list keys request.
 #[derive(Debug, Clone, Serialize)]
 pub struct ListKeysRequest {
+    /// The api id whose keys to list.
     #[serde(rename = "apiId")]
     pub api_id: String,
 
+    /// If provided, this will only return keys where the `ownerId` matches.
     #[serde(rename = "ownerId")]
     pub owner_id: Option<String>,
 
+    /// Limit the number of returned keys, the maximum is 100.
     pub limit: Option<usize>,
 
+    /// An offset for pagination.
     pub offset: Option<usize>,
 }
 
-#[derive(Debug, Clone, Serialize)]
+impl ListKeysRequest {
+    /// Creates a new request for listing keys.
+    ///
+    /// # Arguments
+    /// - `api_id`: The api id to whose keys to list.
+    ///
+    /// # Returns
+    /// - [`KeysListResponse`]: List of `KeysListResponse`.
+    ///
+    /// # Example
+    /// ```
+    /// # use unkey_sdk::models::ListKeysRequest;
+    /// let r = ListKeysRequest::new("test");
+    ///
+    /// assert_eq!(r.api_id, String::from("test"));
+    /// assert_eq!(r.limit, None);
+    /// assert_eq!(r.offset, None);
+    /// assert_eq!(r.owner_id, None);
+    /// ```
+    #[must_use]
+    pub fn new<T: Into<String>>(api_id: T) -> Self {
+        Self {
+            api_id: api_id.into(),
+            owner_id: None,
+            limit: None,
+            offset: None,
+        }
+    }
+
+    /// Sets limit for the keys list request.
+    /// # Arguments
+    /// - `limit`: The limit to set. Default 100.
+    ///
+    /// # Returns
+    /// - [`Self`]: for chained calls.
+    ///
+    /// # Example
+    /// ```
+    /// # use unkey_sdk::models::ListKeysRequest;
+    /// let r = ListKeysRequest::new("test").set_limit(50);
+    ///
+    /// assert_eq!(r.limit.unwrap(), 50);
+    /// ```
+    #[must_use]
+    pub fn set_limit(mut self, limit: usize) -> Self {
+        self.limit = Some(limit);
+        self
+    }
+
+    /// Sets offset for the keys list request.
```

Comment:
```suggestion
    /// Sets the pagination offset for the request.
    ///
```
Same as above, missing new line.